{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Python 3.11.6\n",
      "Requirement already satisfied: pip in ./env/lib64/python3.11/site-packages (23.2.1)\n",
      "Collecting pip\n",
      "  Obtaining dependency information for pip from https://files.pythonhosted.org/packages/47/6a/453160888fab7c6a432a6e25f8afe6256d0d9f2cbd25971021da6491d899/pip-23.3.1-py3-none-any.whl.metadata\n",
      "  Using cached pip-23.3.1-py3-none-any.whl.metadata (3.5 kB)\n",
      "Using cached pip-23.3.1-py3-none-any.whl (2.1 MB)\n",
      "Installing collected packages: pip\n",
      "  Attempting uninstall: pip\n",
      "    Found existing installation: pip 23.2.1\n",
      "    Uninstalling pip-23.2.1:\n",
      "      Successfully uninstalled pip-23.2.1\n",
      "Successfully installed pip-23.3.1\n"
     ]
    }
   ],
   "source": [
    "!python --version\n",
    "!pip install --upgrade pip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "!module load mpi/openmpi-x86_64"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mWARNING: No matching packages for pattern \"mpi4py\"\u001b[0m\u001b[33m\n",
      "\u001b[0mFiles removed: 0\n"
     ]
    }
   ],
   "source": [
    "!python -m pip cache remove mpi4py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting mpi4py\n",
      "  Using cached mpi4py-3.1.5.tar.gz (2.5 MB)\n",
      "  Installing build dependencies ... \u001b[?25ldone\n",
      "\u001b[?25h  Getting requirements to build wheel ... \u001b[?25ldone\n",
      "\u001b[?25h  Preparing metadata (pyproject.toml) ... \u001b[?25ldone\n",
      "\u001b[?25hBuilding wheels for collected packages: mpi4py\n",
      "  Building wheel for mpi4py (pyproject.toml) ... \u001b[?25lerror\n",
      "  \u001b[1;31merror\u001b[0m: \u001b[1msubprocess-exited-with-error\u001b[0m\n",
      "  \n",
      "  \u001b[31m×\u001b[0m \u001b[32mBuilding wheel for mpi4py \u001b[0m\u001b[1;32m(\u001b[0m\u001b[32mpyproject.toml\u001b[0m\u001b[1;32m)\u001b[0m did not run successfully.\n",
      "  \u001b[31m│\u001b[0m exit code: \u001b[1;36m1\u001b[0m\n",
      "  \u001b[31m╰─>\u001b[0m \u001b[31m[227 lines of output]\u001b[0m\n",
      "  \u001b[31m   \u001b[0m running bdist_wheel\n",
      "  \u001b[31m   \u001b[0m running build\n",
      "  \u001b[31m   \u001b[0m running build_src\n",
      "  \u001b[31m   \u001b[0m running build_py\n",
      "  \u001b[31m   \u001b[0m creating build\n",
      "  \u001b[31m   \u001b[0m creating build/lib.linux-x86_64-cpython-311\n",
      "  \u001b[31m   \u001b[0m creating build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/__init__.py -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/__main__.py -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/bench.py -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/run.py -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m creating build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/__init__.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/__main__.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/_base.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/_core.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/_lib.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/aplus.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/pool.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/server.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m creating build/lib.linux-x86_64-cpython-311/mpi4py/util\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/util/__init__.py -> build/lib.linux-x86_64-cpython-311/mpi4py/util\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/util/dtlib.py -> build/lib.linux-x86_64-cpython-311/mpi4py/util\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/util/pkl5.py -> build/lib.linux-x86_64-cpython-311/mpi4py/util\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/MPI.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/__init__.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/__main__.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/bench.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/dl.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/run.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/py.typed -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/MPI.pxd -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/__init__.pxd -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/libmpi.pxd -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m creating build/lib.linux-x86_64-cpython-311/mpi4py/include\n",
      "  \u001b[31m   \u001b[0m creating build/lib.linux-x86_64-cpython-311/mpi4py/include/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/include/mpi4py/mpi4py.MPI.h -> build/lib.linux-x86_64-cpython-311/mpi4py/include/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/include/mpi4py/mpi4py.MPI_api.h -> build/lib.linux-x86_64-cpython-311/mpi4py/include/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/include/mpi4py/mpi4py.h -> build/lib.linux-x86_64-cpython-311/mpi4py/include/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/include/mpi4py/mpi4py.i -> build/lib.linux-x86_64-cpython-311/mpi4py/include/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/include/mpi4py/mpi.pxi -> build/lib.linux-x86_64-cpython-311/mpi4py/include/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/__init__.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/__main__.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/_core.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/_lib.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/aplus.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/pool.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/server.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/util/__init__.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/util\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/util/dtlib.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/util\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/util/pkl5.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/util\n",
      "  \u001b[31m   \u001b[0m running build_clib\n",
      "  \u001b[31m   \u001b[0m MPI configuration: [mpi] from 'mpi.cfg'\n",
      "  \u001b[31m   \u001b[0m MPI C compiler:    /usr/lib64/openmpi/bin/mpiCC\n",
      "  \u001b[31m   \u001b[0m checking for library 'lmpe' ...\n",
      "  \u001b[31m   \u001b[0m /usr/lib64/openmpi/bin/mpiCC -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m The Open MPI wrapper compiler was unable to find the specified compiler\n",
      "  \u001b[31m   \u001b[0m g++ in your PATH.\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m Note that this compiler was either specified at configure time or in\n",
      "  \u001b[31m   \u001b[0m one of several possible environment variables.\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m building 'mpe' dylib library\n",
      "  \u001b[31m   \u001b[0m creating build/temp.linux-x86_64-cpython-311\n",
      "  \u001b[31m   \u001b[0m creating build/temp.linux-x86_64-cpython-311/src\n",
      "  \u001b[31m   \u001b[0m creating build/temp.linux-x86_64-cpython-311/src/lib-pmpi\n",
      "  \u001b[31m   \u001b[0m /usr/lib64/openmpi/bin/mpiCC -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c src/lib-pmpi/mpe.c -o build/temp.linux-x86_64-cpython-311/src/lib-pmpi/mpe.o\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m The Open MPI wrapper compiler was unable to find the specified compiler\n",
      "  \u001b[31m   \u001b[0m g++ in your PATH.\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m Note that this compiler was either specified at configure time or in\n",
      "  \u001b[31m   \u001b[0m one of several possible environment variables.\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m warning: build_clib: command '/usr/lib64/openmpi/bin/mpiCC' failed with exit code 1\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m warning: build_clib: building optional library \"mpe\" failed\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m checking for library 'vt-mpi' ...\n",
      "  \u001b[31m   \u001b[0m /usr/lib64/openmpi/bin/mpiCC -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m The Open MPI wrapper compiler was unable to find the specified compiler\n",
      "  \u001b[31m   \u001b[0m g++ in your PATH.\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m Note that this compiler was either specified at configure time or in\n",
      "  \u001b[31m   \u001b[0m one of several possible environment variables.\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m checking for library 'vt.mpi' ...\n",
      "  \u001b[31m   \u001b[0m /usr/lib64/openmpi/bin/mpiCC -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m The Open MPI wrapper compiler was unable to find the specified compiler\n",
      "  \u001b[31m   \u001b[0m g++ in your PATH.\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m Note that this compiler was either specified at configure time or in\n",
      "  \u001b[31m   \u001b[0m one of several possible environment variables.\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m building 'vt' dylib library\n",
      "  \u001b[31m   \u001b[0m /usr/lib64/openmpi/bin/mpiCC -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c src/lib-pmpi/vt.c -o build/temp.linux-x86_64-cpython-311/src/lib-pmpi/vt.o\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m The Open MPI wrapper compiler was unable to find the specified compiler\n",
      "  \u001b[31m   \u001b[0m g++ in your PATH.\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m Note that this compiler was either specified at configure time or in\n",
      "  \u001b[31m   \u001b[0m one of several possible environment variables.\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m warning: build_clib: command '/usr/lib64/openmpi/bin/mpiCC' failed with exit code 1\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m warning: build_clib: building optional library \"vt\" failed\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m checking for library 'vt-mpi' ...\n",
      "  \u001b[31m   \u001b[0m /usr/lib64/openmpi/bin/mpiCC -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m The Open MPI wrapper compiler was unable to find the specified compiler\n",
      "  \u001b[31m   \u001b[0m g++ in your PATH.\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m Note that this compiler was either specified at configure time or in\n",
      "  \u001b[31m   \u001b[0m one of several possible environment variables.\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m checking for library 'vt.mpi' ...\n",
      "  \u001b[31m   \u001b[0m /usr/lib64/openmpi/bin/mpiCC -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m The Open MPI wrapper compiler was unable to find the specified compiler\n",
      "  \u001b[31m   \u001b[0m g++ in your PATH.\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m Note that this compiler was either specified at configure time or in\n",
      "  \u001b[31m   \u001b[0m one of several possible environment variables.\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m building 'vt-mpi' dylib library\n",
      "  \u001b[31m   \u001b[0m /usr/lib64/openmpi/bin/mpiCC -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c src/lib-pmpi/vt-mpi.c -o build/temp.linux-x86_64-cpython-311/src/lib-pmpi/vt-mpi.o\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m The Open MPI wrapper compiler was unable to find the specified compiler\n",
      "  \u001b[31m   \u001b[0m g++ in your PATH.\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m Note that this compiler was either specified at configure time or in\n",
      "  \u001b[31m   \u001b[0m one of several possible environment variables.\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m warning: build_clib: command '/usr/lib64/openmpi/bin/mpiCC' failed with exit code 1\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m warning: build_clib: building optional library \"vt-mpi\" failed\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m checking for library 'vt-hyb' ...\n",
      "  \u001b[31m   \u001b[0m /usr/lib64/openmpi/bin/mpiCC -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m The Open MPI wrapper compiler was unable to find the specified compiler\n",
      "  \u001b[31m   \u001b[0m g++ in your PATH.\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m Note that this compiler was either specified at configure time or in\n",
      "  \u001b[31m   \u001b[0m one of several possible environment variables.\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m checking for library 'vt.ompi' ...\n",
      "  \u001b[31m   \u001b[0m /usr/lib64/openmpi/bin/mpiCC -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m The Open MPI wrapper compiler was unable to find the specified compiler\n",
      "  \u001b[31m   \u001b[0m g++ in your PATH.\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m Note that this compiler was either specified at configure time or in\n",
      "  \u001b[31m   \u001b[0m one of several possible environment variables.\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m building 'vt-hyb' dylib library\n",
      "  \u001b[31m   \u001b[0m /usr/lib64/openmpi/bin/mpiCC -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c src/lib-pmpi/vt-hyb.c -o build/temp.linux-x86_64-cpython-311/src/lib-pmpi/vt-hyb.o\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m The Open MPI wrapper compiler was unable to find the specified compiler\n",
      "  \u001b[31m   \u001b[0m g++ in your PATH.\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m Note that this compiler was either specified at configure time or in\n",
      "  \u001b[31m   \u001b[0m one of several possible environment variables.\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m warning: build_clib: command '/usr/lib64/openmpi/bin/mpiCC' failed with exit code 1\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m warning: build_clib: building optional library \"vt-hyb\" failed\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m running build_ext\n",
      "  \u001b[31m   \u001b[0m MPI configuration: [mpi] from 'mpi.cfg'\n",
      "  \u001b[31m   \u001b[0m MPI C compiler:    /usr/lib64/openmpi/bin/mpiCC\n",
      "  \u001b[31m   \u001b[0m checking for dlopen() availability ...\n",
      "  \u001b[31m   \u001b[0m checking for header 'dlfcn.h' ...\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -I/home/soap1196/Documents/GitHub/ModinResearch/env/include -I/usr/include/python3.11 -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m success!\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m success!\n",
      "  \u001b[31m   \u001b[0m checking for library 'dl' ...\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -I/home/soap1196/Documents/GitHub/ModinResearch/env/include -I/usr/include/python3.11 -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m gcc _configtest.o -L/usr/lib64 -Lbuild/temp.linux-x86_64-cpython-311 -Wl,--enable-new-dtags,-R/usr/lib64 -ldl -o _configtest\n",
      "  \u001b[31m   \u001b[0m success!\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o _configtest\n",
      "  \u001b[31m   \u001b[0m checking for function 'dlopen' ...\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -I/home/soap1196/Documents/GitHub/ModinResearch/env/include -I/usr/include/python3.11 -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m gcc _configtest.o -L/usr/lib64 -Lbuild/temp.linux-x86_64-cpython-311 -Wl,--enable-new-dtags,-R/usr/lib64 -ldl -o _configtest\n",
      "  \u001b[31m   \u001b[0m success!\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o _configtest\n",
      "  \u001b[31m   \u001b[0m building 'mpi4py.dl' extension\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -DHAVE_DLFCN_H=1 -DHAVE_DLOPEN=1 -I/home/soap1196/Documents/GitHub/ModinResearch/env/include -I/usr/include/python3.11 -c src/dynload.c -o build/temp.linux-x86_64-cpython-311/src/dynload.o\n",
      "  \u001b[31m   \u001b[0m src/dynload.c:5:10: fatal error: Python.h: No such file or directory\n",
      "  \u001b[31m   \u001b[0m     5 | #include \"Python.h\"\n",
      "  \u001b[31m   \u001b[0m       |          ^~~~~~~~~~\n",
      "  \u001b[31m   \u001b[0m compilation terminated.\n",
      "  \u001b[31m   \u001b[0m warning: build_ext: command '/usr/bin/gcc' failed with exit code 1\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m warning: build_ext: building optional extension \"mpi4py.dl\" failed\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m checking for MPI compile and link ...\n",
      "  \u001b[31m   \u001b[0m /usr/lib64/openmpi/bin/mpiCC -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -I/home/soap1196/Documents/GitHub/ModinResearch/env/include -I/usr/include/python3.11 -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m The Open MPI wrapper compiler was unable to find the specified compiler\n",
      "  \u001b[31m   \u001b[0m g++ in your PATH.\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m Note that this compiler was either specified at configure time or in\n",
      "  \u001b[31m   \u001b[0m one of several possible environment variables.\n",
      "  \u001b[31m   \u001b[0m --------------------------------------------------------------------------\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m error: Cannot compile MPI programs. Check your configuration!!!\n",
      "  \u001b[31m   \u001b[0m \u001b[31m[end of output]\u001b[0m\n",
      "  \n",
      "  \u001b[1;35mnote\u001b[0m: This error originates from a subprocess, and is likely not a problem with pip.\n",
      "\u001b[?25h\u001b[31m  ERROR: Failed building wheel for mpi4py\u001b[0m\u001b[31m\n",
      "\u001b[0mFailed to build mpi4py\n",
      "\u001b[31mERROR: Could not build wheels for mpi4py, which is required to install pyproject.toml-based projects\u001b[0m\u001b[31m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!env MPICC=/usr/lib64/openmpi/bin/mpiCC python -m pip install mpi4py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting modin[all]\n",
      "  Using cached modin-0.25.1-py3-none-any.whl.metadata (17 kB)\n",
      "Collecting pandas<2.2,>=2.1 (from modin[all])\n",
      "  Using cached pandas-2.1.4-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (18 kB)\n",
      "Requirement already satisfied: packaging>=21.0 in ./env/lib64/python3.11/site-packages (from modin[all]) (23.2)\n",
      "Collecting numpy>=1.22.4 (from modin[all])\n",
      "  Using cached numpy-1.26.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (61 kB)\n",
      "Collecting fsspec>=2022.05.0 (from modin[all])\n",
      "  Using cached fsspec-2023.12.2-py3-none-any.whl.metadata (6.8 kB)\n",
      "Requirement already satisfied: psutil>=5.8.0 in ./env/lib64/python3.11/site-packages (from modin[all]) (5.9.6)\n",
      "Collecting dask>=2.22.0 (from modin[all])\n",
      "  Using cached dask-2023.12.0-py3-none-any.whl.metadata (3.7 kB)\n",
      "Collecting distributed>=2.22.0 (from modin[all])\n",
      "  Using cached distributed-2023.12.0-py3-none-any.whl.metadata (3.4 kB)\n",
      "Collecting ray!=2.5.0,>=1.13.0 (from ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached ray-2.8.1-cp311-cp311-manylinux2014_x86_64.whl.metadata (13 kB)\n",
      "Collecting pyarrow>=7.0.0 (from modin[all])\n",
      "  Using cached pyarrow-14.0.1-cp311-cp311-manylinux_2_28_x86_64.whl.metadata (3.0 kB)\n",
      "Collecting pydantic<2 (from modin[all])\n",
      "  Using cached pydantic-1.10.13-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (149 kB)\n",
      "Collecting unidist<=0.4.1,>=0.2.1 (from unidist[mpi]<=0.4.1,>=0.2.1; extra == \"all\"->modin[all])\n",
      "  Using cached unidist-0.4.1-py3-none-any.whl.metadata (7.4 kB)\n",
      "Collecting modin-spreadsheet>=0.1.0 (from modin[all])\n",
      "  Using cached modin_spreadsheet-0.1.2-py2.py3-none-any.whl (1.8 MB)\n",
      "Collecting click>=8.1 (from dask>=2.22.0->modin[all])\n",
      "  Using cached click-8.1.7-py3-none-any.whl.metadata (3.0 kB)\n",
      "Collecting cloudpickle>=1.5.0 (from dask>=2.22.0->modin[all])\n",
      "  Using cached cloudpickle-3.0.0-py3-none-any.whl.metadata (7.0 kB)\n",
      "Collecting partd>=1.2.0 (from dask>=2.22.0->modin[all])\n",
      "  Using cached partd-1.4.1-py3-none-any.whl.metadata (4.6 kB)\n",
      "Collecting pyyaml>=5.3.1 (from dask>=2.22.0->modin[all])\n",
      "  Using cached PyYAML-6.0.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.1 kB)\n",
      "Collecting toolz>=0.10.0 (from dask>=2.22.0->modin[all])\n",
      "  Using cached toolz-0.12.0-py3-none-any.whl (55 kB)\n",
      "Collecting importlib-metadata>=4.13.0 (from dask>=2.22.0->modin[all])\n",
      "  Using cached importlib_metadata-7.0.0-py3-none-any.whl.metadata (4.9 kB)\n",
      "Collecting jinja2>=2.10.3 (from distributed>=2.22.0->modin[all])\n",
      "  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)\n",
      "Collecting locket>=1.0.0 (from distributed>=2.22.0->modin[all])\n",
      "  Using cached locket-1.0.0-py2.py3-none-any.whl (4.4 kB)\n",
      "Collecting msgpack>=1.0.0 (from distributed>=2.22.0->modin[all])\n",
      "  Using cached msgpack-1.0.7-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (9.1 kB)\n",
      "Collecting sortedcontainers>=2.0.5 (from distributed>=2.22.0->modin[all])\n",
      "  Using cached sortedcontainers-2.4.0-py2.py3-none-any.whl (29 kB)\n",
      "Collecting tblib>=1.6.0 (from distributed>=2.22.0->modin[all])\n",
      "  Using cached tblib-3.0.0-py3-none-any.whl.metadata (25 kB)\n",
      "Requirement already satisfied: tornado>=6.0.4 in ./env/lib64/python3.11/site-packages (from distributed>=2.22.0->modin[all]) (6.4)\n",
      "Collecting urllib3>=1.24.3 (from distributed>=2.22.0->modin[all])\n",
      "  Using cached urllib3-2.1.0-py3-none-any.whl.metadata (6.4 kB)\n",
      "Collecting zict>=3.0.0 (from distributed>=2.22.0->modin[all])\n",
      "  Using cached zict-3.0.0-py2.py3-none-any.whl (43 kB)\n",
      "Collecting jupyter>=1.0.0 (from modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached jupyter-1.0.0-py2.py3-none-any.whl (2.7 kB)\n",
      "Collecting notebook>=6.0.3 (from modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached notebook-7.0.6-py3-none-any.whl.metadata (10 kB)\n",
      "Collecting ipywidgets>=7.0.0 (from modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached ipywidgets-8.1.1-py3-none-any.whl.metadata (2.4 kB)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in ./env/lib64/python3.11/site-packages (from pandas<2.2,>=2.1->modin[all]) (2.8.2)\n",
      "Collecting pytz>=2020.1 (from pandas<2.2,>=2.1->modin[all])\n",
      "  Using cached pytz-2023.3.post1-py2.py3-none-any.whl.metadata (22 kB)\n",
      "Collecting tzdata>=2022.1 (from pandas<2.2,>=2.1->modin[all])\n",
      "  Using cached tzdata-2023.3-py2.py3-none-any.whl (341 kB)\n",
      "Collecting typing-extensions>=4.2.0 (from pydantic<2->modin[all])\n",
      "  Using cached typing_extensions-4.9.0-py3-none-any.whl.metadata (3.0 kB)\n",
      "Collecting filelock (from ray!=2.5.0,>=1.13.0->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached filelock-3.13.1-py3-none-any.whl.metadata (2.8 kB)\n",
      "Collecting jsonschema (from ray!=2.5.0,>=1.13.0->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached jsonschema-4.20.0-py3-none-any.whl.metadata (8.1 kB)\n",
      "Collecting protobuf!=3.19.5,>=3.15.3 (from ray!=2.5.0,>=1.13.0->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached protobuf-4.25.1-cp37-abi3-manylinux2014_x86_64.whl.metadata (541 bytes)\n",
      "Collecting aiosignal (from ray!=2.5.0,>=1.13.0->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached aiosignal-1.3.1-py3-none-any.whl (7.6 kB)\n",
      "Collecting frozenlist (from ray!=2.5.0,>=1.13.0->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached frozenlist-1.4.0-cp311-cp311-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (5.2 kB)\n",
      "Collecting requests (from ray!=2.5.0,>=1.13.0->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached requests-2.31.0-py3-none-any.whl.metadata (4.6 kB)\n",
      "Collecting aiohttp>=3.7 (from ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached aiohttp-3.9.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.4 kB)\n",
      "Collecting aiohttp-cors (from ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached aiohttp_cors-0.7.0-py3-none-any.whl (27 kB)\n",
      "Collecting colorful (from ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached colorful-0.5.5-py2.py3-none-any.whl (201 kB)\n",
      "Collecting py-spy>=0.2.0 (from ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached py_spy-0.3.14-py2.py3-none-manylinux_2_5_x86_64.manylinux1_x86_64.whl (3.0 MB)\n",
      "Collecting gpustat>=1.0.0 (from ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached gpustat-1.1.1-py3-none-any.whl\n",
      "Collecting opencensus (from ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached opencensus-0.11.3-py2.py3-none-any.whl.metadata (12 kB)\n",
      "Collecting prometheus-client>=0.7.1 (from ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached prometheus_client-0.19.0-py3-none-any.whl.metadata (1.8 kB)\n",
      "Collecting smart-open (from ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached smart_open-6.4.0-py3-none-any.whl.metadata (21 kB)\n",
      "Collecting virtualenv<20.21.1,>=20.0.24 (from ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached virtualenv-20.21.0-py3-none-any.whl (8.7 MB)\n",
      "Collecting grpcio>=1.42.0 (from ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached grpcio-1.60.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.0 kB)\n",
      "INFO: pip is looking at multiple versions of unidist[mpi] to determine which version is compatible with other requirements. This could take a while.\n",
      "Collecting unidist[mpi]<=0.4.1,>=0.2.1 (from modin[all])\n",
      "  Using cached unidist-0.4.0-py3-none-any.whl.metadata (7.4 kB)\n",
      "  Using cached unidist-0.3.0-py3-none-any.whl (104 kB)\n",
      "  Using cached unidist-0.2.2-py3-none-any.whl (102 kB)\n",
      "  Using cached unidist-0.2.1-py3-none-any.whl (102 kB)\n",
      "Collecting modin[all]\n",
      "  Using cached modin-0.25.0-py3-none-any.whl.metadata (17 kB)\n",
      "Collecting unidist>=0.2.1 (from unidist[mpi]>=0.2.1; extra == \"all\"->modin[all])\n",
      "  Using cached unidist-0.5.1-py3-none-manylinux1_x86_64.whl.metadata (7.9 kB)\n",
      "Collecting mpi4py>=3.0.3 (from unidist[mpi]>=0.2.1; extra == \"all\"->modin[all])\n",
      "  Using cached mpi4py-3.1.5.tar.gz (2.5 MB)\n",
      "  Installing build dependencies ... \u001b[?25ldone\n",
      "\u001b[?25h  Getting requirements to build wheel ... \u001b[?25ldone\n",
      "\u001b[?25h  Preparing metadata (pyproject.toml) ... \u001b[?25ldone\n",
      "\u001b[?25hCollecting attrs>=17.3.0 (from aiohttp>=3.7->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached attrs-23.1.0-py3-none-any.whl (61 kB)\n",
      "Collecting multidict<7.0,>=4.5 (from aiohttp>=3.7->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached multidict-6.0.4-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (117 kB)\n",
      "Collecting yarl<2.0,>=1.0 (from aiohttp>=3.7->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached yarl-1.9.4-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (31 kB)\n",
      "Collecting nvidia-ml-py>=11.450.129 (from gpustat>=1.0.0->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached nvidia_ml_py-12.535.133-py3-none-any.whl.metadata (8.6 kB)\n",
      "Collecting blessed>=1.17.1 (from gpustat>=1.0.0->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached blessed-1.20.0-py2.py3-none-any.whl (58 kB)\n",
      "Collecting zipp>=0.5 (from importlib-metadata>=4.13.0->dask>=2.22.0->modin[all])\n",
      "  Using cached zipp-3.17.0-py3-none-any.whl.metadata (3.7 kB)\n",
      "Requirement already satisfied: comm>=0.1.3 in ./env/lib64/python3.11/site-packages (from ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all]) (0.2.0)\n",
      "Requirement already satisfied: ipython>=6.1.0 in ./env/lib64/python3.11/site-packages (from ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all]) (8.18.1)\n",
      "Requirement already satisfied: traitlets>=4.3.1 in ./env/lib64/python3.11/site-packages (from ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all]) (5.14.0)\n",
      "Collecting widgetsnbextension~=4.0.9 (from ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached widgetsnbextension-4.0.9-py3-none-any.whl.metadata (1.6 kB)\n",
      "Collecting jupyterlab-widgets~=3.0.9 (from ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached jupyterlab_widgets-3.0.9-py3-none-any.whl.metadata (4.1 kB)\n",
      "Collecting MarkupSafe>=2.0 (from jinja2>=2.10.3->distributed>=2.22.0->modin[all])\n",
      "  Using cached MarkupSafe-2.1.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.0 kB)\n",
      "Collecting qtconsole (from jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached qtconsole-5.5.1-py3-none-any.whl.metadata (5.1 kB)\n",
      "Collecting jupyter-console (from jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached jupyter_console-6.6.3-py3-none-any.whl (24 kB)\n",
      "Collecting nbconvert (from jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached nbconvert-7.12.0-py3-none-any.whl.metadata (7.7 kB)\n",
      "Requirement already satisfied: ipykernel in ./env/lib64/python3.11/site-packages (from jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all]) (6.27.1)\n",
      "Collecting jupyter-server<3,>=2.4.0 (from notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached jupyter_server-2.12.1-py3-none-any.whl.metadata (8.4 kB)\n",
      "Collecting jupyterlab-server<3,>=2.22.1 (from notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached jupyterlab_server-2.25.2-py3-none-any.whl.metadata (5.9 kB)\n",
      "Collecting jupyterlab<5,>=4.0.2 (from notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached jupyterlab-4.0.9-py3-none-any.whl.metadata (15 kB)\n",
      "Collecting notebook-shim<0.3,>=0.2 (from notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached notebook_shim-0.2.3-py3-none-any.whl (13 kB)\n",
      "Requirement already satisfied: six>=1.5 in ./env/lib64/python3.11/site-packages (from python-dateutil>=2.8.2->pandas<2.2,>=2.1->modin[all]) (1.16.0)\n",
      "Collecting distlib<1,>=0.3.6 (from virtualenv<20.21.1,>=20.0.24->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached distlib-0.3.8-py2.py3-none-any.whl.metadata (5.1 kB)\n",
      "Collecting platformdirs<4,>=2.4 (from virtualenv<20.21.1,>=20.0.24->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached platformdirs-3.11.0-py3-none-any.whl.metadata (11 kB)\n",
      "Collecting jsonschema-specifications>=2023.03.6 (from jsonschema->ray!=2.5.0,>=1.13.0->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached jsonschema_specifications-2023.11.2-py3-none-any.whl.metadata (3.0 kB)\n",
      "Collecting referencing>=0.28.4 (from jsonschema->ray!=2.5.0,>=1.13.0->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached referencing-0.32.0-py3-none-any.whl.metadata (2.7 kB)\n",
      "Collecting rpds-py>=0.7.1 (from jsonschema->ray!=2.5.0,>=1.13.0->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached rpds_py-0.13.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.9 kB)\n",
      "Collecting opencensus-context>=0.1.3 (from opencensus->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached opencensus_context-0.1.3-py2.py3-none-any.whl (5.1 kB)\n",
      "Collecting google-api-core<3.0.0,>=1.0.0 (from opencensus->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached google_api_core-2.15.0-py3-none-any.whl.metadata (2.7 kB)\n",
      "Collecting charset-normalizer<4,>=2 (from requests->ray!=2.5.0,>=1.13.0->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached charset_normalizer-3.3.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (33 kB)\n",
      "Collecting idna<4,>=2.5 (from requests->ray!=2.5.0,>=1.13.0->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached idna-3.6-py3-none-any.whl.metadata (9.9 kB)\n",
      "Collecting certifi>=2017.4.17 (from requests->ray!=2.5.0,>=1.13.0->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached certifi-2023.11.17-py3-none-any.whl.metadata (2.2 kB)\n",
      "Requirement already satisfied: wcwidth>=0.1.4 in ./env/lib64/python3.11/site-packages (from blessed>=1.17.1->gpustat>=1.0.0->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all]) (0.2.12)\n",
      "Collecting googleapis-common-protos<2.0.dev0,>=1.56.2 (from google-api-core<3.0.0,>=1.0.0->opencensus->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached googleapis_common_protos-1.62.0-py2.py3-none-any.whl.metadata (1.5 kB)\n",
      "Collecting google-auth<3.0.dev0,>=2.14.1 (from google-api-core<3.0.0,>=1.0.0->opencensus->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached google_auth-2.25.2-py2.py3-none-any.whl.metadata (4.7 kB)\n",
      "Requirement already satisfied: decorator in ./env/lib64/python3.11/site-packages (from ipython>=6.1.0->ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all]) (5.1.1)\n",
      "Requirement already satisfied: jedi>=0.16 in ./env/lib64/python3.11/site-packages (from ipython>=6.1.0->ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all]) (0.19.1)\n",
      "Requirement already satisfied: matplotlib-inline in ./env/lib64/python3.11/site-packages (from ipython>=6.1.0->ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all]) (0.1.6)\n",
      "Requirement already satisfied: prompt-toolkit<3.1.0,>=3.0.41 in ./env/lib64/python3.11/site-packages (from ipython>=6.1.0->ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all]) (3.0.42)\n",
      "Requirement already satisfied: pygments>=2.4.0 in ./env/lib64/python3.11/site-packages (from ipython>=6.1.0->ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all]) (2.17.2)\n",
      "Requirement already satisfied: stack-data in ./env/lib64/python3.11/site-packages (from ipython>=6.1.0->ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all]) (0.6.3)\n",
      "Requirement already satisfied: pexpect>4.3 in ./env/lib64/python3.11/site-packages (from ipython>=6.1.0->ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all]) (4.9.0)\n",
      "Collecting anyio>=3.1.0 (from jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached anyio-4.1.0-py3-none-any.whl.metadata (4.5 kB)\n",
      "Collecting argon2-cffi (from jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached argon2_cffi-23.1.0-py3-none-any.whl.metadata (5.2 kB)\n",
      "Requirement already satisfied: jupyter-client>=7.4.4 in ./env/lib64/python3.11/site-packages (from jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all]) (8.6.0)\n",
      "Requirement already satisfied: jupyter-core!=5.0.*,>=4.12 in ./env/lib64/python3.11/site-packages (from jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all]) (5.5.0)\n",
      "Collecting jupyter-events>=0.9.0 (from jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached jupyter_events-0.9.0-py3-none-any.whl.metadata (5.7 kB)\n",
      "Collecting jupyter-server-terminals (from jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached jupyter_server_terminals-0.5.0-py3-none-any.whl.metadata (5.6 kB)\n",
      "Collecting nbformat>=5.3.0 (from jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached nbformat-5.9.2-py3-none-any.whl.metadata (3.4 kB)\n",
      "Collecting overrides (from jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached overrides-7.4.0-py3-none-any.whl.metadata (5.7 kB)\n",
      "Requirement already satisfied: pyzmq>=24 in ./env/lib64/python3.11/site-packages (from jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all]) (25.1.2)\n",
      "Collecting send2trash>=1.8.2 (from jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached Send2Trash-1.8.2-py3-none-any.whl (18 kB)\n",
      "Collecting terminado>=0.8.3 (from jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached terminado-0.18.0-py3-none-any.whl.metadata (5.8 kB)\n",
      "Collecting websocket-client (from jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached websocket_client-1.7.0-py3-none-any.whl.metadata (7.9 kB)\n",
      "Collecting async-lru>=1.0.0 (from jupyterlab<5,>=4.0.2->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached async_lru-2.0.4-py3-none-any.whl.metadata (4.5 kB)\n",
      "Collecting jupyter-lsp>=2.0.0 (from jupyterlab<5,>=4.0.2->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached jupyter_lsp-2.2.1-py3-none-any.whl.metadata (1.8 kB)\n",
      "Collecting babel>=2.10 (from jupyterlab-server<3,>=2.22.1->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached Babel-2.14.0-py3-none-any.whl.metadata (1.6 kB)\n",
      "Collecting json5>=0.9.0 (from jupyterlab-server<3,>=2.22.1->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached json5-0.9.14-py2.py3-none-any.whl.metadata (10 kB)\n",
      "Collecting beautifulsoup4 (from nbconvert->jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached beautifulsoup4-4.12.2-py3-none-any.whl (142 kB)\n",
      "Collecting bleach!=5.0.0 (from nbconvert->jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached bleach-6.1.0-py3-none-any.whl.metadata (30 kB)\n",
      "Collecting defusedxml (from nbconvert->jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached defusedxml-0.7.1-py2.py3-none-any.whl (25 kB)\n",
      "Collecting jupyterlab-pygments (from nbconvert->jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached jupyterlab_pygments-0.3.0-py3-none-any.whl.metadata (4.4 kB)\n",
      "Collecting mistune<4,>=2.0.3 (from nbconvert->jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached mistune-3.0.2-py3-none-any.whl.metadata (1.7 kB)\n",
      "Collecting nbclient>=0.5.0 (from nbconvert->jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached nbclient-0.9.0-py3-none-any.whl.metadata (7.8 kB)\n",
      "Collecting pandocfilters>=1.4.1 (from nbconvert->jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached pandocfilters-1.5.0-py2.py3-none-any.whl (8.7 kB)\n",
      "Collecting tinycss2 (from nbconvert->jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached tinycss2-1.2.1-py3-none-any.whl (21 kB)\n",
      "Requirement already satisfied: debugpy>=1.6.5 in ./env/lib64/python3.11/site-packages (from ipykernel->jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all]) (1.8.0)\n",
      "Requirement already satisfied: nest-asyncio in ./env/lib64/python3.11/site-packages (from ipykernel->jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all]) (1.5.8)\n",
      "Collecting qtpy>=2.4.0 (from qtconsole->jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached QtPy-2.4.1-py3-none-any.whl.metadata (12 kB)\n",
      "Collecting sniffio>=1.1 (from anyio>=3.1.0->jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached sniffio-1.3.0-py3-none-any.whl (10 kB)\n",
      "Collecting webencodings (from bleach!=5.0.0->nbconvert->jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached webencodings-0.5.1-py2.py3-none-any.whl (11 kB)\n",
      "Collecting cachetools<6.0,>=2.0.0 (from google-auth<3.0.dev0,>=2.14.1->google-api-core<3.0.0,>=1.0.0->opencensus->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached cachetools-5.3.2-py3-none-any.whl.metadata (5.2 kB)\n",
      "Collecting pyasn1-modules>=0.2.1 (from google-auth<3.0.dev0,>=2.14.1->google-api-core<3.0.0,>=1.0.0->opencensus->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached pyasn1_modules-0.3.0-py2.py3-none-any.whl (181 kB)\n",
      "Collecting rsa<5,>=3.1.4 (from google-auth<3.0.dev0,>=2.14.1->google-api-core<3.0.0,>=1.0.0->opencensus->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached rsa-4.9-py3-none-any.whl (34 kB)\n",
      "Requirement already satisfied: parso<0.9.0,>=0.8.3 in ./env/lib64/python3.11/site-packages (from jedi>=0.16->ipython>=6.1.0->ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all]) (0.8.3)\n",
      "Collecting python-json-logger>=2.0.4 (from jupyter-events>=0.9.0->jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached python_json_logger-2.0.7-py3-none-any.whl (8.1 kB)\n",
      "Collecting rfc3339-validator (from jupyter-events>=0.9.0->jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached rfc3339_validator-0.1.4-py2.py3-none-any.whl (3.5 kB)\n",
      "Collecting rfc3986-validator>=0.1.1 (from jupyter-events>=0.9.0->jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached rfc3986_validator-0.1.1-py2.py3-none-any.whl (4.2 kB)\n",
      "Collecting fastjsonschema (from nbformat>=5.3.0->jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached fastjsonschema-2.19.0-py3-none-any.whl.metadata (2.0 kB)\n",
      "Requirement already satisfied: ptyprocess>=0.5 in ./env/lib64/python3.11/site-packages (from pexpect>4.3->ipython>=6.1.0->ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all]) (0.7.0)\n",
      "Collecting argon2-cffi-bindings (from argon2-cffi->jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached argon2_cffi_bindings-21.2.0-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (86 kB)\n",
      "Collecting soupsieve>1.2 (from beautifulsoup4->nbconvert->jupyter>=1.0.0->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached soupsieve-2.5-py3-none-any.whl.metadata (4.7 kB)\n",
      "Requirement already satisfied: executing>=1.2.0 in ./env/lib64/python3.11/site-packages (from stack-data->ipython>=6.1.0->ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all]) (2.0.1)\n",
      "Requirement already satisfied: asttokens>=2.1.0 in ./env/lib64/python3.11/site-packages (from stack-data->ipython>=6.1.0->ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all]) (2.4.1)\n",
      "Requirement already satisfied: pure-eval in ./env/lib64/python3.11/site-packages (from stack-data->ipython>=6.1.0->ipywidgets>=7.0.0->modin-spreadsheet>=0.1.0->modin[all]) (0.2.2)\n",
      "Collecting fqdn (from jsonschema[format-nongpl]>=4.18.0->jupyter-events>=0.9.0->jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached fqdn-1.5.1-py3-none-any.whl (9.1 kB)\n",
      "Collecting isoduration (from jsonschema[format-nongpl]>=4.18.0->jupyter-events>=0.9.0->jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached isoduration-20.11.0-py3-none-any.whl (11 kB)\n",
      "Collecting jsonpointer>1.13 (from jsonschema[format-nongpl]>=4.18.0->jupyter-events>=0.9.0->jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached jsonpointer-2.4-py2.py3-none-any.whl.metadata (2.5 kB)\n",
      "Collecting uri-template (from jsonschema[format-nongpl]>=4.18.0->jupyter-events>=0.9.0->jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached uri_template-1.3.0-py3-none-any.whl.metadata (8.8 kB)\n",
      "Collecting webcolors>=1.11 (from jsonschema[format-nongpl]>=4.18.0->jupyter-events>=0.9.0->jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached webcolors-1.13-py3-none-any.whl (14 kB)\n",
      "Collecting pyasn1<0.6.0,>=0.4.6 (from pyasn1-modules>=0.2.1->google-auth<3.0.dev0,>=2.14.1->google-api-core<3.0.0,>=1.0.0->opencensus->ray[default]!=2.5.0,>=1.13.0; extra == \"all\"->modin[all])\n",
      "  Using cached pyasn1-0.5.1-py2.py3-none-any.whl.metadata (8.6 kB)\n",
      "Collecting cffi>=1.0.1 (from argon2-cffi-bindings->argon2-cffi->jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached cffi-1.16.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (1.5 kB)\n",
      "Collecting pycparser (from cffi>=1.0.1->argon2-cffi-bindings->argon2-cffi->jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)\n",
      "Collecting arrow>=0.15.0 (from isoduration->jsonschema[format-nongpl]>=4.18.0->jupyter-events>=0.9.0->jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached arrow-1.3.0-py3-none-any.whl.metadata (7.5 kB)\n",
      "Collecting types-python-dateutil>=2.8.10 (from arrow>=0.15.0->isoduration->jsonschema[format-nongpl]>=4.18.0->jupyter-events>=0.9.0->jupyter-server<3,>=2.4.0->notebook>=6.0.3->modin-spreadsheet>=0.1.0->modin[all])\n",
      "  Using cached types_python_dateutil-2.8.19.14-py3-none-any.whl.metadata (1.6 kB)\n",
      "Using cached dask-2023.12.0-py3-none-any.whl (1.2 MB)\n",
      "Using cached distributed-2023.12.0-py3-none-any.whl (997 kB)\n",
      "Using cached fsspec-2023.12.2-py3-none-any.whl (168 kB)\n",
      "Using cached numpy-1.26.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (18.2 MB)\n",
      "Using cached pandas-2.1.4-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.2 MB)\n",
      "Using cached pyarrow-14.0.1-cp311-cp311-manylinux_2_28_x86_64.whl (38.0 MB)\n",
      "Using cached pydantic-1.10.13-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)\n",
      "Using cached ray-2.8.1-cp311-cp311-manylinux2014_x86_64.whl (63.1 MB)\n",
      "Using cached unidist-0.5.1-py3-none-manylinux1_x86_64.whl (206 kB)\n",
      "Using cached modin-0.25.0-py3-none-any.whl (1.1 MB)\n",
      "Using cached aiohttp-3.9.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.3 MB)\n",
      "Using cached click-8.1.7-py3-none-any.whl (97 kB)\n",
      "Using cached cloudpickle-3.0.0-py3-none-any.whl (20 kB)\n",
      "Using cached frozenlist-1.4.0-cp311-cp311-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (250 kB)\n",
      "Using cached grpcio-1.60.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (5.4 MB)\n",
      "Using cached importlib_metadata-7.0.0-py3-none-any.whl (23 kB)\n",
      "Using cached ipywidgets-8.1.1-py3-none-any.whl (139 kB)\n",
      "Using cached msgpack-1.0.7-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (557 kB)\n",
      "Using cached notebook-7.0.6-py3-none-any.whl (4.0 MB)\n",
      "Using cached partd-1.4.1-py3-none-any.whl (18 kB)\n",
      "Using cached prometheus_client-0.19.0-py3-none-any.whl (54 kB)\n",
      "Using cached protobuf-4.25.1-cp37-abi3-manylinux2014_x86_64.whl (294 kB)\n",
      "Using cached pytz-2023.3.post1-py2.py3-none-any.whl (502 kB)\n",
      "Using cached PyYAML-6.0.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (757 kB)\n",
      "Using cached tblib-3.0.0-py3-none-any.whl (12 kB)\n",
      "Using cached typing_extensions-4.9.0-py3-none-any.whl (32 kB)\n",
      "Using cached urllib3-2.1.0-py3-none-any.whl (104 kB)\n",
      "Using cached filelock-3.13.1-py3-none-any.whl (11 kB)\n",
      "Using cached jsonschema-4.20.0-py3-none-any.whl (84 kB)\n",
      "Using cached opencensus-0.11.3-py2.py3-none-any.whl (128 kB)\n",
      "Using cached requests-2.31.0-py3-none-any.whl (62 kB)\n",
      "Using cached smart_open-6.4.0-py3-none-any.whl (57 kB)\n",
      "Using cached certifi-2023.11.17-py3-none-any.whl (162 kB)\n",
      "Using cached charset_normalizer-3.3.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (140 kB)\n",
      "Using cached distlib-0.3.8-py2.py3-none-any.whl (468 kB)\n",
      "Using cached google_api_core-2.15.0-py3-none-any.whl (121 kB)\n",
      "Using cached idna-3.6-py3-none-any.whl (61 kB)\n",
      "Using cached jsonschema_specifications-2023.11.2-py3-none-any.whl (17 kB)\n",
      "Using cached jupyter_server-2.12.1-py3-none-any.whl (380 kB)\n",
      "Using cached jupyterlab-4.0.9-py3-none-any.whl (9.2 MB)\n",
      "Using cached jupyterlab_server-2.25.2-py3-none-any.whl (58 kB)\n",
      "Using cached jupyterlab_widgets-3.0.9-py3-none-any.whl (214 kB)\n",
      "Using cached MarkupSafe-2.1.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (28 kB)\n",
      "Using cached nbconvert-7.12.0-py3-none-any.whl (256 kB)\n",
      "Using cached nvidia_ml_py-12.535.133-py3-none-any.whl (37 kB)\n",
      "Using cached platformdirs-3.11.0-py3-none-any.whl (17 kB)\n",
      "Using cached referencing-0.32.0-py3-none-any.whl (26 kB)\n",
      "Using cached rpds_py-0.13.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.2 MB)\n",
      "Using cached widgetsnbextension-4.0.9-py3-none-any.whl (2.3 MB)\n",
      "Using cached yarl-1.9.4-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (328 kB)\n",
      "Using cached zipp-3.17.0-py3-none-any.whl (7.4 kB)\n",
      "Using cached qtconsole-5.5.1-py3-none-any.whl (123 kB)\n",
      "Using cached anyio-4.1.0-py3-none-any.whl (83 kB)\n",
      "Using cached async_lru-2.0.4-py3-none-any.whl (6.1 kB)\n",
      "Using cached Babel-2.14.0-py3-none-any.whl (11.0 MB)\n",
      "Using cached bleach-6.1.0-py3-none-any.whl (162 kB)\n",
      "Using cached google_auth-2.25.2-py2.py3-none-any.whl (184 kB)\n",
      "Using cached googleapis_common_protos-1.62.0-py2.py3-none-any.whl (228 kB)\n",
      "Using cached json5-0.9.14-py2.py3-none-any.whl (19 kB)\n",
      "Using cached jupyter_events-0.9.0-py3-none-any.whl (18 kB)\n",
      "Using cached jupyter_lsp-2.2.1-py3-none-any.whl (66 kB)\n",
      "Using cached mistune-3.0.2-py3-none-any.whl (47 kB)\n",
      "Using cached nbclient-0.9.0-py3-none-any.whl (24 kB)\n",
      "Using cached nbformat-5.9.2-py3-none-any.whl (77 kB)\n",
      "Using cached QtPy-2.4.1-py3-none-any.whl (93 kB)\n",
      "Using cached terminado-0.18.0-py3-none-any.whl (14 kB)\n",
      "Using cached argon2_cffi-23.1.0-py3-none-any.whl (15 kB)\n",
      "Using cached jupyter_server_terminals-0.5.0-py3-none-any.whl (13 kB)\n",
      "Using cached jupyterlab_pygments-0.3.0-py3-none-any.whl (15 kB)\n",
      "Using cached overrides-7.4.0-py3-none-any.whl (17 kB)\n",
      "Using cached websocket_client-1.7.0-py3-none-any.whl (58 kB)\n",
      "Using cached cachetools-5.3.2-py3-none-any.whl (9.3 kB)\n",
      "Using cached soupsieve-2.5-py3-none-any.whl (36 kB)\n",
      "Using cached fastjsonschema-2.19.0-py3-none-any.whl (23 kB)\n",
      "Using cached cffi-1.16.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (464 kB)\n",
      "Using cached jsonpointer-2.4-py2.py3-none-any.whl (7.8 kB)\n",
      "Using cached pyasn1-0.5.1-py2.py3-none-any.whl (84 kB)\n",
      "Using cached uri_template-1.3.0-py3-none-any.whl (11 kB)\n",
      "Using cached arrow-1.3.0-py3-none-any.whl (66 kB)\n",
      "Using cached types_python_dateutil-2.8.19.14-py3-none-any.whl (9.4 kB)\n",
      "Building wheels for collected packages: mpi4py\n",
      "  Building wheel for mpi4py (pyproject.toml) ... \u001b[?25lerror\n",
      "  \u001b[1;31merror\u001b[0m: \u001b[1msubprocess-exited-with-error\u001b[0m\n",
      "  \n",
      "  \u001b[31m×\u001b[0m \u001b[32mBuilding wheel for mpi4py \u001b[0m\u001b[1;32m(\u001b[0m\u001b[32mpyproject.toml\u001b[0m\u001b[1;32m)\u001b[0m did not run successfully.\n",
      "  \u001b[31m│\u001b[0m exit code: \u001b[1;36m1\u001b[0m\n",
      "  \u001b[31m╰─>\u001b[0m \u001b[31m[155 lines of output]\u001b[0m\n",
      "  \u001b[31m   \u001b[0m running bdist_wheel\n",
      "  \u001b[31m   \u001b[0m running build\n",
      "  \u001b[31m   \u001b[0m running build_src\n",
      "  \u001b[31m   \u001b[0m running build_py\n",
      "  \u001b[31m   \u001b[0m creating build\n",
      "  \u001b[31m   \u001b[0m creating build/lib.linux-x86_64-cpython-311\n",
      "  \u001b[31m   \u001b[0m creating build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/__init__.py -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/__main__.py -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/bench.py -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/run.py -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m creating build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/__init__.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/__main__.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/_base.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/_core.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/_lib.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/aplus.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/pool.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/server.py -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m creating build/lib.linux-x86_64-cpython-311/mpi4py/util\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/util/__init__.py -> build/lib.linux-x86_64-cpython-311/mpi4py/util\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/util/dtlib.py -> build/lib.linux-x86_64-cpython-311/mpi4py/util\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/util/pkl5.py -> build/lib.linux-x86_64-cpython-311/mpi4py/util\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/MPI.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/__init__.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/__main__.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/bench.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/dl.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/run.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/py.typed -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/MPI.pxd -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/__init__.pxd -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/libmpi.pxd -> build/lib.linux-x86_64-cpython-311/mpi4py\n",
      "  \u001b[31m   \u001b[0m creating build/lib.linux-x86_64-cpython-311/mpi4py/include\n",
      "  \u001b[31m   \u001b[0m creating build/lib.linux-x86_64-cpython-311/mpi4py/include/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/include/mpi4py/mpi4py.MPI.h -> build/lib.linux-x86_64-cpython-311/mpi4py/include/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/include/mpi4py/mpi4py.MPI_api.h -> build/lib.linux-x86_64-cpython-311/mpi4py/include/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/include/mpi4py/mpi4py.h -> build/lib.linux-x86_64-cpython-311/mpi4py/include/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/include/mpi4py/mpi4py.i -> build/lib.linux-x86_64-cpython-311/mpi4py/include/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/include/mpi4py/mpi.pxi -> build/lib.linux-x86_64-cpython-311/mpi4py/include/mpi4py\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/__init__.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/__main__.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/_core.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/_lib.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/aplus.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/pool.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/futures/server.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/futures\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/util/__init__.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/util\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/util/dtlib.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/util\n",
      "  \u001b[31m   \u001b[0m copying src/mpi4py/util/pkl5.pyi -> build/lib.linux-x86_64-cpython-311/mpi4py/util\n",
      "  \u001b[31m   \u001b[0m running build_clib\n",
      "  \u001b[31m   \u001b[0m MPI configuration: [mpi] from 'mpi.cfg'\n",
      "  \u001b[31m   \u001b[0m checking for library 'lmpe' ...\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m gcc _configtest.o -llmpe -o _configtest\n",
      "  \u001b[31m   \u001b[0m /usr/bin/ld: cannot find -llmpe: No such file or directory\n",
      "  \u001b[31m   \u001b[0m collect2: error: ld returned 1 exit status\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m building 'mpe' dylib library\n",
      "  \u001b[31m   \u001b[0m creating build/temp.linux-x86_64-cpython-311\n",
      "  \u001b[31m   \u001b[0m creating build/temp.linux-x86_64-cpython-311/src\n",
      "  \u001b[31m   \u001b[0m creating build/temp.linux-x86_64-cpython-311/src/lib-pmpi\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c src/lib-pmpi/mpe.c -o build/temp.linux-x86_64-cpython-311/src/lib-pmpi/mpe.o\n",
      "  \u001b[31m   \u001b[0m creating build/lib.linux-x86_64-cpython-311/mpi4py/lib-pmpi\n",
      "  \u001b[31m   \u001b[0m gcc -shared -Wl,--no-as-needed build/temp.linux-x86_64-cpython-311/src/lib-pmpi/mpe.o -o build/lib.linux-x86_64-cpython-311/mpi4py/lib-pmpi/libmpe.so\n",
      "  \u001b[31m   \u001b[0m checking for library 'vt-mpi' ...\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m gcc _configtest.o -lvt-mpi -o _configtest\n",
      "  \u001b[31m   \u001b[0m /usr/bin/ld: cannot find -lvt-mpi: No such file or directory\n",
      "  \u001b[31m   \u001b[0m collect2: error: ld returned 1 exit status\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m checking for library 'vt.mpi' ...\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m gcc _configtest.o -lvt.mpi -o _configtest\n",
      "  \u001b[31m   \u001b[0m /usr/bin/ld: cannot find -lvt.mpi: No such file or directory\n",
      "  \u001b[31m   \u001b[0m collect2: error: ld returned 1 exit status\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m building 'vt' dylib library\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c src/lib-pmpi/vt.c -o build/temp.linux-x86_64-cpython-311/src/lib-pmpi/vt.o\n",
      "  \u001b[31m   \u001b[0m gcc -shared -Wl,--no-as-needed build/temp.linux-x86_64-cpython-311/src/lib-pmpi/vt.o -o build/lib.linux-x86_64-cpython-311/mpi4py/lib-pmpi/libvt.so\n",
      "  \u001b[31m   \u001b[0m checking for library 'vt-mpi' ...\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m gcc _configtest.o -lvt-mpi -o _configtest\n",
      "  \u001b[31m   \u001b[0m /usr/bin/ld: cannot find -lvt-mpi: No such file or directory\n",
      "  \u001b[31m   \u001b[0m collect2: error: ld returned 1 exit status\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m checking for library 'vt.mpi' ...\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m gcc _configtest.o -lvt.mpi -o _configtest\n",
      "  \u001b[31m   \u001b[0m /usr/bin/ld: cannot find -lvt.mpi: No such file or directory\n",
      "  \u001b[31m   \u001b[0m collect2: error: ld returned 1 exit status\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m building 'vt-mpi' dylib library\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c src/lib-pmpi/vt-mpi.c -o build/temp.linux-x86_64-cpython-311/src/lib-pmpi/vt-mpi.o\n",
      "  \u001b[31m   \u001b[0m gcc -shared -Wl,--no-as-needed build/temp.linux-x86_64-cpython-311/src/lib-pmpi/vt-mpi.o -o build/lib.linux-x86_64-cpython-311/mpi4py/lib-pmpi/libvt-mpi.so\n",
      "  \u001b[31m   \u001b[0m checking for library 'vt-hyb' ...\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m gcc _configtest.o -lvt-hyb -o _configtest\n",
      "  \u001b[31m   \u001b[0m /usr/bin/ld: cannot find -lvt-hyb: No such file or directory\n",
      "  \u001b[31m   \u001b[0m collect2: error: ld returned 1 exit status\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m checking for library 'vt.ompi' ...\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m gcc _configtest.o -lvt.ompi -o _configtest\n",
      "  \u001b[31m   \u001b[0m /usr/bin/ld: cannot find -lvt.ompi: No such file or directory\n",
      "  \u001b[31m   \u001b[0m collect2: error: ld returned 1 exit status\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m building 'vt-hyb' dylib library\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -c src/lib-pmpi/vt-hyb.c -o build/temp.linux-x86_64-cpython-311/src/lib-pmpi/vt-hyb.o\n",
      "  \u001b[31m   \u001b[0m gcc -shared -Wl,--no-as-needed build/temp.linux-x86_64-cpython-311/src/lib-pmpi/vt-hyb.o -o build/lib.linux-x86_64-cpython-311/mpi4py/lib-pmpi/libvt-hyb.so\n",
      "  \u001b[31m   \u001b[0m running build_ext\n",
      "  \u001b[31m   \u001b[0m MPI configuration: [mpi] from 'mpi.cfg'\n",
      "  \u001b[31m   \u001b[0m checking for dlopen() availability ...\n",
      "  \u001b[31m   \u001b[0m checking for header 'dlfcn.h' ...\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -I/home/soap1196/Documents/GitHub/ModinResearch/env/include -I/usr/include/python3.11 -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m success!\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m success!\n",
      "  \u001b[31m   \u001b[0m checking for library 'dl' ...\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -I/home/soap1196/Documents/GitHub/ModinResearch/env/include -I/usr/include/python3.11 -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m gcc _configtest.o -L/usr/lib64 -Lbuild/temp.linux-x86_64-cpython-311 -Wl,--enable-new-dtags,-R/usr/lib64 -ldl -o _configtest\n",
      "  \u001b[31m   \u001b[0m success!\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o _configtest\n",
      "  \u001b[31m   \u001b[0m checking for function 'dlopen' ...\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -I/home/soap1196/Documents/GitHub/ModinResearch/env/include -I/usr/include/python3.11 -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m gcc _configtest.o -L/usr/lib64 -Lbuild/temp.linux-x86_64-cpython-311 -Wl,--enable-new-dtags,-R/usr/lib64 -ldl -o _configtest\n",
      "  \u001b[31m   \u001b[0m success!\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o _configtest\n",
      "  \u001b[31m   \u001b[0m building 'mpi4py.dl' extension\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -DHAVE_DLFCN_H=1 -DHAVE_DLOPEN=1 -I/home/soap1196/Documents/GitHub/ModinResearch/env/include -I/usr/include/python3.11 -c src/dynload.c -o build/temp.linux-x86_64-cpython-311/src/dynload.o\n",
      "  \u001b[31m   \u001b[0m src/dynload.c:5:10: fatal error: Python.h: No such file or directory\n",
      "  \u001b[31m   \u001b[0m     5 | #include \"Python.h\"\n",
      "  \u001b[31m   \u001b[0m       |          ^~~~~~~~~~\n",
      "  \u001b[31m   \u001b[0m compilation terminated.\n",
      "  \u001b[31m   \u001b[0m warning: build_ext: command '/usr/bin/gcc' failed with exit code 1\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m warning: build_ext: building optional extension \"mpi4py.dl\" failed\n",
      "  \u001b[31m   \u001b[0m \n",
      "  \u001b[31m   \u001b[0m checking for MPI compile and link ...\n",
      "  \u001b[31m   \u001b[0m gcc -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -fcf-protection -fexceptions -fcf-protection -fexceptions -fcf-protection -fexceptions -fPIC -I/home/soap1196/Documents/GitHub/ModinResearch/env/include -I/usr/include/python3.11 -c _configtest.c -o _configtest.o\n",
      "  \u001b[31m   \u001b[0m _configtest.c:2:10: fatal error: mpi.h: No such file or directory\n",
      "  \u001b[31m   \u001b[0m     2 | #include <mpi.h>\n",
      "  \u001b[31m   \u001b[0m       |          ^~~~~~~\n",
      "  \u001b[31m   \u001b[0m compilation terminated.\n",
      "  \u001b[31m   \u001b[0m failure.\n",
      "  \u001b[31m   \u001b[0m removing: _configtest.c _configtest.o\n",
      "  \u001b[31m   \u001b[0m error: Cannot compile MPI programs. Check your configuration!!!\n",
      "  \u001b[31m   \u001b[0m \u001b[31m[end of output]\u001b[0m\n",
      "  \n",
      "  \u001b[1;35mnote\u001b[0m: This error originates from a subprocess, and is likely not a problem with pip.\n",
      "\u001b[?25h\u001b[31m  ERROR: Failed building wheel for mpi4py\u001b[0m\u001b[31m\n",
      "\u001b[0mFailed to build mpi4py\n",
      "\u001b[31mERROR: Could not build wheels for mpi4py, which is required to install pyproject.toml-based projects\u001b[0m\u001b[31m\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!pip install \"modin[all]\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import modin.pandas as pd\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
